app/models/concerns/foreman_puppet_enc/puppet_lookup_value_extensions.rb

```diff
@@ -0,0 +1,13 @@
+module ForemanPuppetEnc
+  module PuppetLookupValueExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      validate :value_present?, if: ->(v) { v.lookup_key.is_a?(ForemanPuppetEnc::PuppetclassLookupKey) }
```

Comment:
maybe some other distinction of type would be appropriate ... :thinking:
But that would need to be in core.